src/outputs/restart_hdf5.cpp

```diff
@@ -102,6 +103,12 @@ RestartReaderHDF5::SparseInfo RestartReaderHDF5::GetSparseInfo() const {
     // Read data from file
     PARTHENON_HDF5_CHECK(H5Dread(hdl.dataset, hdl.type, memspace, hdl.dataspace,
                                  H5P_DEFAULT, static_cast<void *>(info.allocated.get())));
+    info.dealloc_count.resize(hdl_dealloc.count);
+    PARTHENON_HDF5_CHECK(H5Dread(hdl_dealloc.dataset, hdl_dealloc.type,
+                                 H5S::FromHIDCheck(H5Screate_simple(
+                                     hdl_dealloc.rank, hdl_dealloc.dims.data(), NULL)),
+                                 hdl_dealloc.dataspace, H5P_DEFAULT,
+                                 static_cast<void *>(info.dealloc_count.data())));
```

Comment:
Do you see a case where this might cause issues for your downstream codes (e.g., loading older restart files)?
If so, does it make sense/is it worth to add a default override to support older files?

We don't have sparse variables yet, so it's up to you.

Reply:
I think Riot is the only code using sparse variables and I don't think we have any legacy data that we need to restart from currently, so I will just leave it as is. This way people will know if they are trying to restart from data that will not give exact results, and it is easy enough to add a workaround in the future if we really need to.